src/components/API.js

```diff
@@ -0,0 +1,83 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _validatePromise(res) {
```

Comment:
It's great to have a function for checking the server response, so this logic is not duplicated